### ISSUE TITLE
Input elements "name" attribute should allow arbitrary strings instead of only ID

### ISSUE THREAD

**Issue by jhorstmann** [axodotdev/axohtml#40]
It's been a long time since I did any html, but I remember being able to use names such as `foo[0].bar` or `foo[]` as input field names, so that a web framework could reconstruct a nested structure from query/form parameters. It seems this is not currently supported, and [the attribute is defined as being of type `ID`](https://github.com/axodotdev/axohtml/blob/main/typed-html/src/elements.rs#L233).

Using such a name leads to a runtime panic such as

```
ERROR: <input name="foobar[]"> failed to parse attribute value: ID can only contain alphanumerics, dash, dot and underscore
```

The [whatwg html spec](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#attr-fe-name) does not seem to restrict the contents of the `name` attribute:

> Other than `isindex`, any non-empty value for `name` is allowed

**Comment by manyinsects** [comment]
This is true, yeah. You'd especially use this in the context of complex web forms that are generated either server-side or by some client-side framework (especially the former is important for `axohtml`'s usage). I think we can relax these restrictions.

**Comment by jhorstmann** [comment]
Thank you for the quick reaction! I'd be happy to provide a PR if you are interested.

**Comment by manyinsects** [comment]
Sure, if you're up for it!
